fastparquet/compression.py

```diff
@@ -33,50 +33,67 @@ def gzip_decompress_v2(data):
 else:
     def gzip_compress_v3(data, compresslevel=COMPRESSION_LEVEL):
         return gzip.compress(data, compresslevel=compresslevel)
+    def gzip_decompress(data, uncompressed_size):
+        return gzip.decompress(data)
     compressions['GZIP'] = gzip_compress_v3
-    decompressions['GZIP'] = gzip.decompress
+    decompressions['GZIP'] = gzip_decompress
 
 try:
     import snappy
+    def snappy_decompress(data, uncompressed_size):
+        return snappy.decompress(data)
     compressions['SNAPPY'] = snappy.compress
-    decompressions['SNAPPY'] = snappy.decompress
+    decompressions['SNAPPY'] = snappy_decompress
 except ImportError:
     pass
 try:
     import lzo
+    def lzo_decompress(data, uncompressed_size):
+        return lzo.decompress(data)
     compressions['LZO'] = lzo.compress
-    decompressions['LZO'] = lzo.decompress
+    decompressions['LZO'] = lzo_decompress
 except ImportError:
     pass
 try:
     import brotli
+    def brotli_decompress(data, uncompressed_size):
+        return brotli.decompress(data)
     compressions['BROTLI'] = brotli.compress
-    decompressions['BROTLI'] = brotli.decompress
+    decompressions['BROTLI'] = brotli_decompress
 except ImportError:
     pass
 try:
-    import lz4.frame
-    compressions['LZ4'] = lz4.frame.compress
-    decompressions['LZ4'] = lz4.frame.decompress
+    import lz4.block
+    def lz4_compress(data, **kwargs):
+        try:
+            if kwargs['store_size'] == True:
+                raise RuntimeError(
+                    'store_size cannot be True for LZ4 the compressor'
+                )
+        except KeyError:
+            kwargs['store_size'] = False
+        return lz4.block.compress(data, **kwargs)
+    def lz4_decompress(data, uncompressed_size):
+        return lz4.block.decompress(data, uncompressed_size=uncompressed_size)
+    compressions['LZ4'] = lz4_compress
+    decompressions['LZ4'] = lz4_decompress
 except ImportError:
     pass
 try:
     import zstd
     def zstd_compress(data, **kwargs):
-        # For the ZstdDecompressor to work, the compressed data must include
-        # the uncompressed size, so we raise an exception if the user tries to
-        # set this to False. We also set it to True if it's not specified
-        # (since the default is False, weirdly).
         try:
-            if kwargs['write_content_size'] == False:
-                raise RuntimeError('write_content_size cannot be false for the ZSTD compressor')
+            if kwargs['write_content_size'] == True:
+                raise RuntimeError(
+                    'write_content_size cannot be false for the ZSTD compressor'
```

Comment:
but `'write_content_size` is `True` here?

Reply:
Oops, typo in the exception message.